code/game/machinery/recharger.dm

```diff
@@ -112,6 +112,7 @@
 			var/obj/item/weapon/gun/energy/E = charging
 			if(E.power_supply.charge < E.power_supply.maxcharge)
 				E.power_supply.give(E.power_supply.chargerate * recharge_coeff)
+				E.recharge_newshot()

```

Comment:
since we no longer reload the chamber when pulling the trigger, we need to try to reload it whenever the energy gun's cell is recharged.